app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -120,7 +124,7 @@ export default class List extends PureComponent {
         }
 
         sections.push({
-            action: this.goToMoreChannels,
+            action: canCreatePublicChannels ? this.goToMoreChannels : null,
```

Comment:
Doesn't hiding this also prevent you from joining other public channels?

Reply:
Yes it will, need to hide the option to create public channela inside the opened modal after the + sign is pressed